lib/iris/aux_factory.py

```diff
@@ -784,20 +758,51 @@ def dependencies(self):
                     depth_c=self.depth_c, nsigma=self.nsigma, zlev=self.zlev)
 
     def _derive(self, sigma, eta, depth, depth_c,
-                zlev, shape, nsigma_slice):
+                zlev, nsigma, coord_dims_func):
+        # Calculate the index of the 'z' dimension in the inputs.
+        # Get the cube dimension...
+        i_levels_cubedim, = coord_dims_func(self.dependencies['zlev'])
+        # ...subtract any previous dimensions not used in the calculation.
+        derived_cubedims = self.derived_dims(coord_dims_func)
+        i_levels_dim = i_levels_cubedim - sum(
+            i_dim not in derived_cubedims
+            for i_dim in range(i_levels_cubedim))
+
+        # Calculate the result shape as a combination of all the inputs.
+        allshapes = np.array(
+            [el.shape
+             for el in (sigma, eta, depth, depth_c, zlev)
+             if el.ndim])
+        result_shape = list(np.max(allshapes, axis=0))
```

Comment:
Yes, I believe it ***is*** provided by `_nd_points` and `_nd_bounds` that the results have all-same dimensions, given they are called with the same 'ndim'.

I will try to amend some docstrings (`_nd_xxx` and `_remap_xxx`) to make this clearer : it's obvious that `_remap_xxx` should have docstrings reallly ...